kore/src/Kore/Simplify/Equals.hs

```diff
@@ -289,27 +295,27 @@ makeEvaluate
     second@Conditional{term = secondTerm}
     sideCondition =
         do
-            let first' = first{term = if termsAreEqual then mkTop_ else firstTerm}
-            firstCeil <- makeEvaluateCeil sideCondition first'
-            let second' = second{term = if termsAreEqual then mkTop_ else secondTerm}
-            secondCeil <- makeEvaluateCeil sideCondition second'
-            firstCeilNegation <- Not.simplifyEvaluated sideCondition firstCeil
-            secondCeilNegation <- Not.simplifyEvaluated sideCondition secondCeil
+            let termSort = termLikeSort firstTerm
```

Comment:
Is there a difference between `termSort` and `sort` from the where-clause? Otherwise, I'd suggest keeping only one definition.